src/regedit/register.jl

```diff
@@ -160,18 +160,23 @@ subset of the pool of available versions, this function computes a `VersionSpec`
 includes all versions in `subset` and none of the versions in its complement.
 """
 function compress_versions(pool::Vector{VersionNumber}, subset::Vector{VersionNumber})
-    subset = sort(subset) # must copy, we mutate this
+    # Explicitly drop prerelease/build numbers, as those can confuse this.
+    # TODO: Rewrite all this to use VersionNumbers instead of VersionBounds
+    drop_build_prerelease(v::VersionNumber) = VersionNumber(v.major, v.minor, v.patch)
+    pool = drop_build_prerelease.(pool)
+    subset = sort!(drop_build_prerelease.(subset))
+
     complement = sort!(setdiff(pool, subset))
     ranges = VersionRange[]
     @label again
     isempty(subset) && return VersionSpec(ranges)
     a = first(subset)
     for b in reverse(subset)
         a.major == b.major || continue
-        for m = 1:3
-            lo = VersionBound((a.major, a.minor, a.patch)[1:m]...)
-            for n = 1:3
-                hi = VersionBound((b.major, b.minor, b.patch)[1:n]...)
+        for m = 1:4
+            lo = VersionBound((a.major, a.minor, a.patch, a.build)[1:m]...)
+            for n = 1:4
+                hi = VersionBound((b.major, b.minor, b.patch, b.build)[1:n]...)
```

Comment:
This is a bug: if `VersionNumber` doesn't have `build` info, then `x.build` is an empty tuple, but there is no  method for `VersionBound` that accepts empty tuples. I have a situation right now where that crashes things. @staticfloat could we either revert this PR in the short run to unblock the situation I'm trying to solve, or come up with a fix?